quickwit-search/src/fetch_docs.rs

```diff
@@ -149,18 +149,16 @@ async fn fetch_docs_in_split<'a>(
     split: &SplitIdAndFooterOffsets,
 ) -> anyhow::Result<Vec<(GlobalDocAddress<'a>, String)>> {
     let index_reader = get_searcher_for_split(global_doc_addrs.len(), index_storage, split).await?;
-    let mut doc_futures = Vec::new();
-    for global_doc_addr in global_doc_addrs {
+    let doc_futures = global_doc_addrs.into_iter().map(|global_doc_addr| {
```

Comment:
`futures::future::try_join_all` accepts any iterable so often, when it does not hurt readability, it's better to pass an iterator rather than needlessly allocating a new vector.

Reply:
good point. Another remark on the error side, I found the usage `.context("searcher-doc-async")?` or `.with_context(|| "searcher-doc-async")?;` too poor for debugging purpose.

Reply:
Yeah, those are usually not helpful without additional context.